index.js

```diff
@@ -61,17 +62,28 @@ var WinstonCloudWatch = function(options) {
 
 util.inherits(WinstonCloudWatch, winston.Transport);
 
-WinstonCloudWatch.prototype.log = function(level, msg, meta, callback) {
-  debug('log (called by winston)', level, msg, meta);
-
-  var log = { level: level, msg: msg, meta: meta };
-  if (!_.isEmpty(msg)) {
+WinstonCloudWatch.prototype.log = function (level, msg, meta, callback) {
+  // Winston 3.x support
+  // `callback` is the second of two parameters: (info, callback)
+  // @see https://github.com/winstonjs/winston#adding-custom-transports
+  var cb = callback === undefined ? msg : callback;
+
+  var log = callback === undefined ?
+  // Winston 3.x support
+  // `level` is an `info` object containing the actual message
+  // @see https://github.com/winstonjs/logform#info-objects
+    { level: level.level, msg: level.message } :
+    { level: level, msg: msg, meta: meta };
+  
+  debug('log (called by winston)', log.level, log.msg, log.meta);
+
+  if (!isEmpty(log.msg)) {
     this.add(log);
   }
 
-  if (!/^uncaughtException: /.test(msg)) {
+  if (!/^uncaughtException: /.test(log.msg)) {
```

Comment:
What's the difference here between using `msg` and `log.msg`?

Reply:
@lazywithclass `msg` var no longer holds the actual message string in Winston 3. So to keep it consistent, I'm using the generated `log` object and made `log.msg` point to what we need in both versions.

Reply:
I see, thanks.